From 817ca7af271374c83cc81f3d64ad2007db97263c Mon Sep 17 00:00:00 2001
From: Shivananda Hebbar <x0hebbar@ti.com>
Date: Fri, 19 Mar 2010 15:37:56 -0600
Subject: [PATCH 072/135] DSPBRIDGE: Change dw_type to u8

channel type can take value 1 or 2.
Hence changing the type to u8.

Signed-off-by: Shivananda Hebbar <x0hebbar@ti.com>
---
 arch/arm/plat-omap/include/dspbridge/_chnl_sm.h |    2 +-
 arch/arm/plat-omap/include/dspbridge/cfg.h      |    4 ++--
 arch/arm/plat-omap/include/dspbridge/chnlpriv.h |    2 +-
 drivers/dsp/bridge/services/cfg.c               |    4 ++--
 4 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/arm/plat-omap/include/dspbridge/_chnl_sm.h b/arch/arm/plat-omap/include/dspbridge/_chnl_sm.h
index bc09ee0..4cea2cc 100644
--- a/arch/arm/plat-omap/include/dspbridge/_chnl_sm.h
+++ b/arch/arm/plat-omap/include/dspbridge/_chnl_sm.h
@@ -128,7 +128,7 @@ struct chnl_mgr {
 	u32 max_channels;	/* Total number of channels */
 	u32 open_channels;	/* Total number of open channels */
 	struct chnl_object **ap_channel;	/* Array of channels */
-	u32 dw_type;		/* Type of channel class library */
+	u8 dw_type;		/* Type of channel class library */
 	/* If no shm syms, return for CHNL_Open */
 	dsp_status chnl_open_status;
 };
diff --git a/arch/arm/plat-omap/include/dspbridge/cfg.h b/arch/arm/plat-omap/include/dspbridge/cfg.h
index a760fdd..1b67660 100644
--- a/arch/arm/plat-omap/include/dspbridge/cfg.h
+++ b/arch/arm/plat-omap/include/dspbridge/cfg.h
@@ -175,7 +175,7 @@ extern dsp_status cfg_get_host_resources(IN struct cfg_devnode *dev_node_obj,
  *      DSP_SOK:    *pdwValue is set to the retrieved u32(non-Zero).
  *      else:       *pdwValue is set to 0L.
  */
-extern dsp_status cfg_get_object(OUT u32 *pdwValue, u32 dw_type);
+extern dsp_status cfg_get_object(OUT u32 *pdwValue, u8 dw_type);
 
 /*
  *  ======== cfg_get_perf_value ========
@@ -286,6 +286,6 @@ extern dsp_status cfg_set_dev_object(IN struct cfg_devnode *dev_node_obj,
  *  Ensures:
  *      DSP_SOK:        The Private u32 was successfully set.
  */
-extern dsp_status cfg_set_object(IN u32 dwValue, IN u32 dw_type);
+extern dsp_status cfg_set_object(IN u32 dwValue, u8 dw_type);
 
 #endif /* CFG_ */
diff --git a/arch/arm/plat-omap/include/dspbridge/chnlpriv.h b/arch/arm/plat-omap/include/dspbridge/chnlpriv.h
index 750b791..8f5e9f4 100644
--- a/arch/arm/plat-omap/include/dspbridge/chnlpriv.h
+++ b/arch/arm/plat-omap/include/dspbridge/chnlpriv.h
@@ -87,7 +87,7 @@ struct chnl_info {
 
 /* Channel manager info: */
 struct chnl_mgrinfo {
-	u32 dw_type;		/* Type of channel class library. */
+	u8 dw_type;		/* Type of channel class library. */
 	/* Channel handle, given the channel id. */
 	struct chnl_object *chnl_obj;
 	u32 open_channels;	/* Number of open channels. */
diff --git a/drivers/dsp/bridge/services/cfg.c b/drivers/dsp/bridge/services/cfg.c
index b1daf27..dae7410 100644
--- a/drivers/dsp/bridge/services/cfg.c
+++ b/drivers/dsp/bridge/services/cfg.c
@@ -208,7 +208,7 @@ dsp_status cfg_get_host_resources(struct cfg_devnode *dev_node_obj,
  *  Purpose:
  *      Retrieve the Object handle from the Registry
  */
-dsp_status cfg_get_object(OUT u32 *pdwValue, u32 dw_type)
+dsp_status cfg_get_object(OUT u32 *pdwValue, u8 dw_type)
 {
 	dsp_status status = -EINVAL;
 	u32 dw_buf_size;
@@ -296,7 +296,7 @@ dsp_status cfg_set_dev_object(struct cfg_devnode *dev_node_obj, u32 dwValue)
  *  Purpose:
  *      Store the Driver Object handle
  */
-dsp_status cfg_set_object(u32 dwValue, u32 dw_type)
+dsp_status cfg_set_object(u32 dwValue, u8 dw_type)
 {
 	dsp_status status = -EINVAL;
 	u32 dw_buff_size;
-- 
1.7.0.4

